src/bootstrap/step.rs

```diff
@@ -814,6 +814,9 @@ invalid rule dependency graph detected, was a rule added and maybe typo'd?
             Subcommand::Clean => panic!(),
         };
 
+        // Only construct our plan from build triple steps for dist builds.
+        let from_build_only = kind == Kind::Dist;
```

Comment:
I originally did something like you suggested, but was faced with borrow checker rage saying the vector slice or anything like that doesn't outlive `'a` up there in the fn sig. Maybe I was just not trying harder? Anyway I'll try to figure it out this time...